Adds warnings but allows no windows, no roofs, no ground floor and no…

mlauster opened the pull request:
… inner walls

closes #423

PRemmen commented on teaser/data/output/modelicatemplate/AixLib/AixLib_ThermalZoneRecord_FourElement:
```diff
@@ -15,7 +15,7 @@ record ${zone.parent.name}_${zone.name} "${zone.parent.name}_${zone.name}"
     RWin = ${zone.model_attr.r1_win},
     gWin = ${zone.model_attr.weighted_g_value},
-    UWin= ${zone.model_attr.ua_value_win/zone.model_attr.area_win},
+    UWin= ${zone.model_attr.ua_value_win},
```
should this be `zone.model_attr.u_value_win` instead of `zone.model_attr.ua_value_win`

mlauster replied:
Fixed

PRemmen commented on teaser/data/output/modelicatemplate/AixLib/AixLib_ThermalZoneRecord_OneElement:
```diff
@@ -14,7 +14,7 @@ record ${zone.parent.name}_${zone.name} "${zone.parent.name}_${zone.name}"
     RWin = ${zone.model_attr.r1_win},
     gWin = ${zone.model_attr.weighted_g_value},
-    UWin= ${zone.model_attr.ua_value_win/zone.model_attr.area_win},
+    UWin= ${zone.model_attr.ua_value_win},
```
see above

mlauster replied:
Fixed

PRemmen commented on teaser/data/output/modelicatemplate/AixLib/AixLib_ThermalZoneRecord_ThreeElement:
```diff
@@ -14,7 +14,7 @@ record ${zone.parent.name}_${zone.name} "${zone.parent.name}_${zone.name}"
     RWin = ${zone.model_attr.r1_win},
     gWin = ${zone.model_attr.weighted_g_value},
-    UWin= ${zone.model_attr.ua_value_win/zone.model_attr.area_win},
+    UWin= ${zone.model_attr.ua_value_win},
```
see above

mlauster replied:
Fixed

PRemmen commented on teaser/data/output/modelicatemplate/AixLib/AixLib_ThermalZoneRecord_TwoElement:
```diff
@@ -14,7 +14,7 @@ record ${zone.parent.name}_${zone.name} "${zone.parent.name}_${zone.name}"
     RWin = ${zone.model_attr.r1_win},
     gWin = ${zone.model_attr.weighted_g_value},
-    UWin= ${zone.model_attr.ua_value_win/zone.model_attr.area_win},
+    UWin= ${zone.model_attr.ua_value_win},
```
see above

mlauster replied:
Fixed

PRemmen commented on teaser/data/output/modelicatemplate/Annex60/Annex60_FourElements:
```diff
@@ -24,7 +24,7 @@ model ${zone.parent.name}_${zone.name}
     annotation (Placement(transformation(extent={{-68,52},{-48,72}})));
   Annex60.ThermalZones.ReducedOrder.SolarGain.CorrectionGDoublePane corGDoublePane(n=${zone.model_attr.n_outer},
-  UWin=${zone.model_attr.ua_value_win/zone.model_attr.area_win})
+  UWin=${zone.model_attr.ua_value_win})
```
see above

mlauster replied:
Fixed

PRemmen commented on teaser/data/output/modelicatemplate/Annex60/Annex60_OneElement:
```diff
@@ -24,7 +24,7 @@ model ${zone.parent.name}_${zone.name}
     annotation (Placement(transformation(extent={{-68,52},{-48,72}})));
   Annex60.ThermalZones.ReducedOrder.SolarGain.CorrectionGDoublePane corGDoublePane(n=${zone.model_attr.n_outer},
-  UWin=${zone.model_attr.ua_value_win/zone.model_attr.area_win})
+  UWin=${zone.model_attr.ua_value_win})
```
see above

mlauster replied:
Fixed

PRemmen commented on teaser/data/output/modelicatemplate/Annex60/Annex60_ThreeElements:
```diff
@@ -24,7 +24,7 @@ model ${zone.parent.name}_${zone.name}
     annotation (Placement(transformation(extent={{-68,52},{-48,72}})));
   Annex60.ThermalZones.ReducedOrder.SolarGain.CorrectionGDoublePane corGDoublePane(n=${zone.model_attr.n_outer},
-  UWin=${zone.model_attr.ua_value_win/zone.model_attr.area_win})
+  UWin=${zone.model_attr.ua_value_win})
```
see above

mlauster replied:
Fixed

PRemmen commented on teaser/data/output/modelicatemplate/Annex60/Annex60_TwoElements:
```diff
@@ -24,7 +24,7 @@ model ${zone.parent.name}_${zone.name}
     annotation (Placement(transformation(extent={{-68,52},{-48,72}})));
   Annex60.ThermalZones.ReducedOrder.SolarGain.CorrectionGDoublePane corGDoublePane(n=${zone.model_attr.n_outer},
-  UWin=${zone.model_attr.ua_value_win/zone.model_attr.area_win})
+  UWin=${zone.model_attr.ua_value_win})
```
see above

mlauster replied:
Fixed

PRemmen commented on teaser/logic/buildingobjects/calculation/four_element.py:
```diff
@@ -288,6 +288,8 @@ class FourElement(object):
     ua_value_win : float [W/K]
         U-Value times outer wall area.
+    u_value_win : float [W/(m2K)]
+        U-Value of windows.
```
My understanding is, that this is the area-weighted U value of all windows. If that is the meaning of this value we might want to explain that.

mlauster replied:
I fixed that by mentioning it in description. Do you think that is sufficient or did you have in mind to rename that parameter?

PRemmen commented on teaser/logic/buildingobjects/calculation/four_element.py:
```diff
@@ -530,6 +532,7 @@ def __init__(self, thermal_zone, merge_windows, t_bt):
 
         # UA-Value
         self.ua_value_win = 0.0
+        self.u_value_win = 0.0
```
although for this value there shouldn't be any difference, but we should also add this value to set_calc_default function of each calculation module

mlauster replied:
Fixed

PRemmen commented on teaser/logic/buildingobjects/calculation/one_element.py:
```diff
@@ -141,7 +141,9 @@ class OneElement(object):
-        U-Value times outer wall area.
+        U-Value times window wall area.
+    u_value_win : float [W/(m2K)]
+        U-Value of windows.
```
see comments at four_element.py

mlauster replied:
Fixed

PRemmen commented on teaser/logic/buildingobjects/calculation/one_element.py:
```diff
@@ -278,6 +280,7 @@ def __init__(self, thermal_zone, merge_windows, t_bt):
 
         # UA-Value
         self.ua_value_win = 0.0
+        self.u_value_win = 0.0
```
see comments at four_element.py

mlauster replied:
Fixed

PRemmen commented on teaser/logic/buildingobjects/calculation/three_element.py:
```diff
@@ -217,6 +217,8 @@ class ThreeElement(object):
     ua_value_win : float [W/K]
         U-Value times outer wall area.
+    u_value_win : float [W/(m2K)]
+        U-Value of windows.
```
see comments at four_element.py

mlauster replied:
Fixed

PRemmen commented on teaser/logic/buildingobjects/calculation/three_element.py:
```diff
@@ -410,6 +412,7 @@ def __init__(self, thermal_zone, merge_windows, t_bt):
 
         # UA-Value
         self.ua_value_win = 0.0
+        self.u_value_win = 0.0
```
see comments at four_element.py

mlauster replied:
Fixed

PRemmen commented on teaser/logic/buildingobjects/calculation/two_element.py:
```diff
@@ -177,7 +177,9 @@ class TwoElement(object):
-        U-Value times outer wall area.
+        U-Value times window wall area.
+    u_value_win : float [W/(m2K)]
+        U-Value of windows.
```
see comments at four_element.py

mlauster replied:
Fixed

PRemmen commented on teaser/logic/buildingobjects/calculation/two_element.py:
```diff
@@ -341,6 +343,7 @@ def __init__(self, thermal_zone, merge_windows, t_bt):
 
         # UA-Value
         self.ua_value_win = 0.0
+        self.u_value_win = 0.0
```
see comments at four_element.py

mlauster replied:
Fixed

mlauster commented:
@PRemmen, I fixed all your comments. I added "are weighted" to description of the U-value. Did you have this in mind or would you like to rename the parameter to make it more clear?
